x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpSslExporterIT.java

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.monitoring.exporter.http;
+
+import org.elasticsearch.action.ActionFuture;
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.elasticsearch.common.settings.MockSecureSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.env.TestEnvironment;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESIntegTestCase.Scope;
+import org.elasticsearch.test.http.MockWebServer;
+import org.elasticsearch.xpack.core.ssl.TestsSSLService;
+import org.elasticsearch.xpack.core.ssl.VerificationMode;
+import org.elasticsearch.xpack.monitoring.exporter.Exporter;
+import org.elasticsearch.xpack.monitoring.exporter.Exporters;
+import org.elasticsearch.xpack.monitoring.test.MonitoringIntegTestCase;
+import org.hamcrest.CoreMatchers;
+import org.junit.AfterClass;
+import org.junit.Before;
+
+import javax.net.ssl.SSLContext;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.notNullValue;
+
+@ESIntegTestCase.ClusterScope(scope = Scope.SUITE,
+    numDataNodes = 1, numClientNodes = 0, transportClientRatio = 0.0, supportsDedicatedMasters = false)
+public class HttpSslExporterIT extends MonitoringIntegTestCase {
+
+    private final Settings globalSettings = Settings.builder().put("path.home", createTempDir()).build();
+    private final Environment environment = TestEnvironment.newEnvironment(globalSettings);
+
+    private static MockWebServer webServer;
+    private static Path keystore;
+    private MockSecureSettings secureSettings;
+
+
+    @AfterClass
+    public static void cleanUpStatics() {
+        if (webServer != null) {
+            webServer.close();
+            webServer = null;
+        }
+        keystore = null;
+    }
+
+    @Override
+    protected boolean ignoreExternalCluster() {
+        return true;
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        if (keystore == null) {
```

Comment:
@jkakavas is there something that should be done here for the FIPS work?

Reply:
Yes, thanks for catching this. 

@tvernum can you please add the key and certificate files and use those instead of the keystore when setting up the [MockWebServer](https://github.com/elastic/elasticsearch/pull/30953/files/e8d82994a211a80af08b9ac6225c7e64df2f7097#diff-18dcff2030412c3efeae23dd8d0df6d4R99) ?
Using it for the truststore is fine, even in a FIPS-140 JVM so no additional requirements here. When I went through our tests, I opted for using the certificate for `certificate_authorities` instead of the JKS for `truststore` (when applicable and made sense ) mainly for consistency, but we don't _have to_ .

Reply:
Oh FIPS why do you want to make my life complicated?